### ISSUE TITLE
Intermediate Atlantis Upgrade Scheduling Call

### ISSUE THREAD

**Issue by soc1c** [ethereumclassic/ECIPs#80]
It became apparent that certain parts of the Ethereum Classic community are appreciating an [accelerated hardfork schedule](https://medium.com/ethereum-classic-labs/etc-labs-upgrades-ethereum-classic-to-expand-functionality-and-improve-compatability-with-ethereum-e2ac7d9aad93). To avoid confusion, rushing protocol upgrades, and putting the network at risk of a potential split, I propose scheduling an intermediate atlantis-upgrade scheduling call to discuss and agree on a realistic timeline for ECIP-1054

ref #79 

### ETC Core Devs Call - Atlantis Scheduling

When: **Friday, June 07, 2019, 1pm UTC**, 30 minutes max.

Where: **Youtube Livestream** https://www.youtube.com/watch?v=hDVrKDSOCWA&feature=youtu.be
You can stream the call and chat and ask questions on Youtube Livestream. If you want access to the Hangouts to chat via voice, you'll need to message @soc1c or @YazzyYaz for a link to that. Due to limit on voice call attendees by Google, we ask that only those who are interested in using voice to ask for the invite to the Hangouts.
### Agenda

- Quick client teams check-in
  - Geth / Multi-Geth 
  - Parity Ethereum
  - IOHK Mantis
- Discuss timeline for the protocol upgrade
    - Morden Classic and Kotti Classic testnet
    - Ethereum Classic mainnet 

There are two competing proposals right now, I will give them letters, and propose a 3rd as a compromise:

- [**A**] Original ECIP-1054 schedule
  - `1_039_000` on Kotti Classic PoA-testnet (early August 2019)
  - `4_723_000` on Morden Classic PoW-testnet (early August 2019)
  - `8_750_000` on Ethereum Classic PoW-mainnet (mid-September 2019)
- [**B**] ETCLabs "July 1st" schedule
  - `N/A` on Kotti Classic PoA-testnet
  - immediately on Morden Classic PoW-testnet (early June 2019)
  - `8_343_000` on Ethereum Classic PoW-mainnet (July 1st 2019)
- [**C**] Moderately accelerated schedule
  - `716_617` on Kotti Classic PoA-testnet (two weeks from now)
  - `4_729_274` on Morden Classic PoW-testnet (two weeks from now)
  - `8_500_000` on Ethereum Classic PoW-mainnet (six weeks after successfull testnet forks)

**Comment by wired1924** [comment]
Where is the link for the call?

**Comment by meowsbits** [comment]
Typo: `[B]` should `s/June/July/`

**Comment by YazzyYaz** [comment]
> Typo: `[B]` should `s/June 1/July 1/`

I updated it, thanks for pointing it out.

**Comment by meowsbits** [comment]
re: `[B]` again, "immediately" is obviously incomplete and impossible. What is the tangible proposed number? (If there is one?)

**Comment by cats9000** [comment]
> Ask for invite here if you are not on that discord.

@soc1c could I please have an invitation?

**Comment by TokenHash** [comment]
1pm UTC is 9am EST right (just to confirm and not to miss it)

**Comment by mikeyb** [comment]
http://www.timebie.com/std/utc.php

**Comment by soc1c** [comment]
@wired1924 @cats9000 https://discord.gg/HJfne6e

**Comment by YazzyYaz** [comment]
After talks with @soc1c @mikeyb and @TokenHash I have updated the ECIP discussion location to the Youtube Livestream here: https://www.youtube.com/watch?v=hDVrKDSOCWA&feature=youtu.be

**Comment by tzdybal** [comment]
Unfortunately I'm not able to participate in the call, but I want to share my opinion as a developer.

**I fully endorse option A - Original ECIP schedule.**

According to my experience with previous forks, it's impossible to avoid network split with tight schedule. In my opinion even the original schedule leave just enough time for informational campaign and software update of the nodes. Because of the decentralized nature of the network, where you can't directly contact all participants, updates take time.

I would also like to remind all parties, that regardless of what will be decided, changes has to be implemented thru ECIP process. In my opinion, change of a activation block is such a significant change, that ECIP should go back to "Draft" state and then wait 2 weeks in "Last call".

**Comment by soc1c** [comment]
Thanks for everyone, I updated the ECIP according to the results of the call in #82

**Comment by YazzyYaz** [comment]
Livestream of Discussion: https://www.youtube.com/watch?v=hDVrKDSOCWA

**Comment by zmitton** [comment]
My understanding is that we are tentatively targeting a compromise date at block 8,500,000 (early august). 

Is this correct? Does anyone have a pressing objection to this? thumbs up, thumbs down?

**Comment by TokenHash** [comment]
The agreed dates with Terry Culver's feedback are:

716_617 on Kotti Classic PoA-testnet (two weeks from now)
4_729_274 on Morden Classic PoW-testnet (two weeks from now)
8_500_000 on Ethereum Classic PoW-mainnet (six weeks after successfull testnet forks)

[the above is from #ecips channel posted by @stevanlohja]

**Comment by soc1c** [comment]
> The agreed dates with Terry Culver's feedback are:
> 
> 716_617 on Kotti Classic PoA-testnet (two weeks from now)
> 4_729_274 on Morden Classic PoW-testnet (two weeks from now)
> 8_500_000 on Ethereum Classic PoW-mainnet (six weeks after successfull testnet forks)
> 
> [the above is from #ecips channel posted by @stevanlohja]

Thanks

**Comment by zmitton** [comment]
with regard to eip161 @meowsbits there is a contract used by the EF to delete _some_ empties. I got it from Martin Swende. [here it is](https://etherscan.io/address/0xa43ebd8939d8328f5858119a3fb65f65c864c6dd#contracts), but it only cleans the ones from the attack, not all of them. My understanding is that the attacker must have generated addresses in a predictable mathematical way, and this contract just erases those ones using his same math. we can give this a try, hopefully it will remove the vast majority. It is also possible to remove all of them, but its a bit difficult to find them due to the "secure tree" abstraction - the fact that  a key use in the tree is the hash of the address. I can find the keys, but then I need thei preimages. I can probably get those another way but would have to comb through every transaction.

tldr: the cleanup happens after the hard fork, so we should move forward, either way. plus i have the contract that can probably do a very good job

**Comment by zmitton** [comment]
the repo for ethereum's state clearing process eip161 is forked into my repo [here](https://github.com/zmitton/statesweep)

if anyone wants to get involved please subscribe or open issues there

**Comment by devfdn** [comment]
what successful testing means for atlantis time schedule?? i am reeading geth code and see this https://github.com/ethereumproject/go-ethereum/issues/717!! can somebodey please explain
